backend/src/test/java/de/wirvsvirus/trackyourbed/HospitalServiceTest.java

```diff
@@ -0,0 +1,349 @@
+      final UUID hospitalId = UUID.randomUUID();
+
+      when(hospitalRepository.findById(hospitalId)).thenReturn(Optional.of(hospital));
+      when(hospitalDtoMapper.entityToDto(hospital)).thenReturn(expected);
+
+      //When
+      HospitalDto actual = hospitalService.getHospitalById(hospitalId);
+
+      //then
+      assertSame(expected, actual);
+      verify(hospitalRepository, times(1)).findById(hospitalId);
+      verify(hospitalDtoMapper, times(1)).entityToDto(hospital);
+    }
+
+    @Test
+    @DisplayName("Should throw exception when no hospital found by id")
+    void shouldThrowNoSuchHospitalExceptionWhenGetHospitalById() {
+      //Given
+      final UUID hospitalId = UUID.randomUUID();
+
+      when(hospitalRepository.findById(hospitalId)).thenReturn(Optional.empty());
+
+      //When
+      NoSuchHospitalException e = assertThrows(
+          NoSuchHospitalException.class,
+          () -> hospitalService.getHospitalById(hospitalId));
+
+      //then
+      assertEquals("Could not find hospital with ID " + hospitalId, e.getMessage());
+      verify(hospitalRepository, times(1)).findById(hospitalId);
+      verifyNoInteractions(hospitalDtoMapper);
+    }
+  }
+
+  @Nested
+  @DisplayName("Test calls to getAllHospitals")
+  class GetAllHospitalsTest {
+
+    @Test
+    @DisplayName("Should return all hospitals.")
+    void shouldGetAllHospitals() {
+      //Given
+      String hospitalOneName = "hos1";
+      String hospitalTwoName = "hos2";
+      final Hospital hospital1 = new Hospital();
+      hospital1.setName(hospitalOneName);
+      hospital1.setMaxCapacity(1);
+      final Hospital hospital2 = new Hospital();
+      hospital2.setName(hospitalTwoName);
+      hospital2.setMaxCapacity(2);
+
+      List<Hospital> hospitals = Arrays.asList(hospital1, hospital2);
+
+      when(hospitalRepository.findAll()).thenReturn(hospitals);
+      when(hospitalDtoMapper.entityToDto(any(Hospital.class))).thenAnswer(arguments -> {
+        HospitalDto hospitalDto = new HospitalDto();
+        hospitalDto.setName(arguments.getArgument(0, Hospital.class).getName());
+        return hospitalDto;
+      });
+      final List<String> expectedHospitalNames = List.of(hospitalOneName, hospitalTwoName);
+
+      //when
+      Collection<HospitalDto> actual = hospitalService.getAllHospitals();
+
+      //then
+      assertEquals(2, actual.size());
+      assertThat(
+          actual.stream().map(HospitalDto::getName).collect(Collectors.toList()),
+          containsInAnyOrder(expectedHospitalNames.toArray())
+      );
+
+      verify(hospitalRepository).findAll();
+      verify(hospitalDtoMapper)
+          .entityToDto(argThat(h -> Objects.equals(hospitalOneName, h.getName())));
+      verify(hospitalDtoMapper)
+          .entityToDto(argThat(h -> Objects.equals(hospitalTwoName, h.getName())));
+    }
+  }
+
+
+  @Nested
+  @DisplayName("Test calls to updateHospital")
+  class updateHospitalTest {
```

Comment:
There is some behaviour that also should be tested:
- if fields are not set in the update hospital request, then they are untouched in the entity (Probably having one test per field is a bit too much, but one with `updateHospital=new UpdateHospital()` and nothing else set would be nice)